src/services/codefixes/fixUnusedIdentifier.ts

```diff
@@ -82,12 +86,16 @@ namespace ts.codefix {
         },
     });
 
+    function hasDeletedAncestor(deletedAncestors: ReadonlyNodeSet, node: Node): boolean {

```

Comment:
can you add a test where the ranges are nested:

```ts
function f(a: any, unused: any) { a; }

f(1, {
    prop: f(2, [
        f(3, f(4, undefined))
    ])
});
```

Reply:
or even more interesting if the first one is the nested reference:
```ts
function f(a: any, unused: any) { a; }
function g(a: any, unused: any) { a; }

g(1, {
    prop: f(2, [
        g(3, f(4, undefined))
    ])
});
```

Reply:
After testing more, looks like the previous strategy wouldn't work because it relied on us deleting higher-leve nodes before lower-level ones. Changed this to just collect a list of deletions and do them all at the end, when we can know what all the high-level deletions are.